Document Operator installation on minikube

dharmit opened the pull request:
**What type of PR is this?**
/kind documentation
[skip ci]

**What does this PR do / why we need it**:
This PR documents steps to install Operators on minikube cluster. We need this to help Kubernetes users get up to speed with installing and working with Operators.

**Which issue(s) this PR fixes**:

Fixes #4103

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [x] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

dharmit commented:
@sbose78 @Avni-Sharma @pedjak can one of you please take a look at the section that has the steps to install SBO on minikube and provide feedback? It's titled as `Installing the Service Binding Operator`.

Avni-Sharma commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,204 @@
+== Installing Operators on minkube
```
nit  - minikube

dharmit replied:
Fixed.

Avni-Sharma commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,204 @@
+subjects:
+- kind: ServiceAccount
+  name: service-binding-operator
+  namespace: default
```
We can add a note that for the example we are using a default ns, this can be changed to any other ns as well.

dharmit replied:
Added it at the top of the section for SBO installation. PTAL and let me know your thoughts.

Avni-Sharma commented:
Rest steps look good to me :+1:

sbose78 commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,204 @@
+$ git clone https://github.com/redhat-developer/service-binding-operator/
+$ cd service-binding-operator
+$ git fetch --tags
+$ git checkout tags/v0.1.1-364 -b v0.1.1-364
```
Could we replace these with a single long yaml file ? Otherwise, for an `odo` user, this is a lot of work :( 
Ideally, every release in Service Binding should have a "long" yaml as a release artifact. 

@pedjak @Avni-Sharma

kadel replied:
@sbose78 @pedjak Would it make sense for SBO to provide one big yaml file (or multiple files if you prefer) with all resources that are required for SBO installation? The users could then just do one simple `kubect create -f ` and get the whole setup?
A good example is how OLM does it https://github.com/operator-framework/operator-lifecycle-manager/releases/tag/0.16.1

sbose78 replied:
+1 @kadel , I would prefer that too.

pedjak replied:
I am on +1 on that as well, but before publishing such yaml file, we should run our acceptance tests to verify if the operator is going to work really. We are working on making our acceptance tests runnable agains plain k8s clusters. cc @pmacik

pedjak commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,204 @@
+
+We use the link:https://github.com/redhat-developer/service-binding-operator/[Service Binding Operator] to provide `odo link` feature which links an odo component with an Operator backed service. Thus, to be able to use this feature, it's essential that we install the Operator first.
+
+Service Binding Operator is not yet available via the OLM. The team is link:https://github.com/redhat-developer/service-binding-operator/issues/727[working on making it available] through OLM.
```
Actually it is available, but not in upstream index, instead in `registry.redhat.io/redhat/community-operator-index:latest`

Hence, we could define the following catalogsource:

```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: community-operators
  namespace: operators
spec:
  displayName: Community Operators
  image: registry.redhat.io/redhat/community-operator-index:latest
  publisher: Red Hat
  sourceType: grpc
  priority: -400
  publisher: Red Hat
  updateStrategy:
    registryPoll:
      interval: 10m0s
```

And then add subscription:

```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: service-binding-operator
  namespace: operators
spec:
  channel: 'beta'
  installPlanApproval: Automatic
  name: service-binding-operator
  source: community-operators
  sourceNamespace: operators
```

That should install the operator.

dharmit replied:
I tried this on a fresh minikube cluster after enabling olm - `minikube addons enable olm`. But the Opertor doesn't show up. Here's `kubectl describe` for the catsrc:
```
Name:         community-operators
Namespace:    operators
Labels:       <none>
Annotations:  API Version:  operators.coreos.com/v1alpha1
Kind:         CatalogSource
Metadata:
  Creation Timestamp:  2020-10-28T06:00:19Z
  Generation:          1
  Managed Fields:
    API Version:  operators.coreos.com/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:kubectl.kubernetes.io/last-applied-configuration:
      f:spec:
        .:
        f:displayName:
        f:image:
        f:publisher:
        f:sourceType:
        f:updateStrategy:
          .:
          f:registryPoll:
            .:
            f:interval:
    Manager:      kubectl
    Operation:    Update
    Time:         2020-10-28T06:00:19Z
    API Version:  operators.coreos.com/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:spec:
        f:icon:
          .:
          f:base64data:
          f:mediatype:
      f:status:
        .:
        f:connectionState:
          .:
          f:address:
          f:lastConnect:
          f:lastObservedState:
        f:registryService:
          .:
          f:createdAt:
          f:port:
          f:protocol:
          f:serviceName:
          f:serviceNamespace:
    Manager:         catalog
    Operation:       Update
    Time:            2020-10-28T06:06:48Z
  Resource Version:  1614
  Self Link:         /apis/operators.coreos.com/v1alpha1/namespaces/operators/catalogsources/community-operators
  UID:               5b722962-5886-4295-a165-59d7b79d2d8d
Spec:
  Display Name:  Community Operators
  Image:         registry.redhat.io/redhat/community-operator-index:latest
  Publisher:     Red Hat
  Source Type:   grpc
  Update Strategy:
    Registry Poll:
      Interval:  10m0s
Status:
  Connection State:
    Address:              community-operators.operators.svc:50051
    Last Connect:         2020-10-28T06:06:48Z
    Last Observed State:  TRANSIENT_FAILURE
  Registry Service:
    Created At:         2020-10-28T06:00:19Z
    Port:               50051
    Protocol:           grpc
    Service Name:       community-operators
    Service Namespace:  operators
Events:                 <none>
```
I think the issue with pulling the image:
```
$ docker pull registry.redhat.io/redhat/community-operator-index:latest
Error response from daemon: Get https://registry.redhat.io/v2/redhat/community-operator-index/manifests/latest: unauthorized: Please login to the Red Hat Registry using your Customer Portal credentials. Further instructions can be found here: https://access.redhat.com/RegistryAuthentication
```
Is it possible to host the image on quay and provide that instead?

Avni-Sharma replied:
Indeed @dharmit  @pedjak we get an ImagePullBackOff err
`operators     community-operators-6r6q4                   0/1     ImagePullBackOff   0          110s
`

pedjak replied:
Yeah, we need to add credentials for that repo into the cluster.

Here is what I did:

* Follow https://access.redhat.com/RegistryAuthentication and create account if you do not have any
* Verify that your credentials are correct using `docker login -u <your_username> -p <you_passwd> registry.redhat.io`
* start minikube cluster
* enable OLM with `minikube addons enable olm` or install it with `curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/0.16.1/install.sh | bash -s 0.16.1`
* create a new image pull secret out of your local `.docker/config.json` file:

```
kubectl create secret generic community-operators-secrets -n olm --from-file=.dockerconfigjson=~/.docker/config.json --type=kubernetes.io/dockerconfigjson
```
* add that pull secret to the default account in `olm` namespace

```
kubectl patch serviceaccount default -p '{"imagePullSecrets": [{"name": "community-operators-secrets"}]}'
```

* add catalog source to `olm` namespace

```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: community-operators
  namespace: olm
spec:
  displayName: Community Operators
  image: registry.redhat.io/redhat/community-operator-index:latest
  publisher: Red Hat
  sourceType: grpc
  publisher: Red Hat
  updateStrategy:
    registryPoll:
      interval: 10m0s
```
* eventually `kubectl get packagemanifest -n olm` will list service-binding-operator as well
* add subscription to `operators` namespace:

```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: service-binding-operator
  namespace: operators
spec:
  channel: 'beta'
  installPlanApproval: Automatic
  name: service-binding-operator
  source: community-operators
  sourceNamespace: olm
```

The operator will eventually be up and running

dharmit replied:
@pedjak I'm making this comment without validating above steps because I'm quite sure that it will work. However, the point is to be able to bring down the barrier to use SBO on minikube/k8s environments.

I'm not sure if users would appreciate the idea of having to create an account on RH Customer Portal to just use an Operator. If we could manage to address https://github.com/redhat-developer/service-binding-operator/issues/727 and make SBO easily available, that would really help.

If I see this as an end user (an enterprise developer who only cares about writing java/nodejs apps that connect to an external service), there's a tonne of stuff I need to do to run Operators and link them with an odo component (`Deployment` in k8s/OCP lingo).

@kadel @sbose78 WDYT?

pedjak replied:
@dharmit we definitelly have a plan to publish the operator on upstream operatorhub - creating account is just a temp thing until this does not happen.

kadel replied:
Most of the upstream Kubernetes won't have RH account. We want to be upstream friendly and RH account requirement is blocker for upstream docs.

pedjak replied:
> Most of the upstream Kubernetes won't have RH account. We want to be upstream friendly and RH account requirement is blocker for upstream docs.

Sure @kadel - I cannot agree more. I have demonstrated above only what is currently available, given that we still do not publish to upstream operatorhub. However, that's our near term plan, but in order that this can happen, we need to first verify if our acceptance tests are green on plain k8s clusters. cc @pmacik

pedjak commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,204 @@
+At the moment, to make Service Binding Operator work on Kubernetes, we need to keep a couple of things in mind:
+
+. It needs to be installed by cloning its GitHub repository
+. It must be installed in an individual namespace unlike the cluster wide installation we saw for etcd Operator in previous section
```
the operator can be installed in all namespaces.

dharmit replied:
How? Is there a doc for this on SBO's repo or website that a user can refer to?

pedjak replied:
That is the OLM default behaviour, i.e. this is what you get if installed using Subscription (see above). If you want to tell operator to watch only some namespaces, then we need to configure via [`OperatorGroup`](https://docs.openshift.com/container-platform/4.6/operators/understanding/olm/olm-understanding-operatorgroups.html)

dharmit replied:
OK. Maybe a silly question but, for below Operator Group, if I install SBO using the CSV [available from here](https://github.com/operator-framework/community-operators/blob/master/community-operators/service-binding-operator/0.1.1-364/service-binding-operator.v0.1.1-364.clusterserviceversion.yaml), I should be able to see SBO getting installed across all the namespaces, right?
```yaml
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"operators.coreos.com/v1","kind":"OperatorGroup","metadata":{"annotations":{},"name":"global-operators","namespace":"operators"}}
    olm.providedAPIs: EtcdBackup.v1beta2.etcd.database.coreos.com,EtcdCluster.v1beta2.etcd.database.coreos.com,EtcdRestore.v1beta2.etcd.database.coreos.com
  name: global-operators
  namespace: operators
spec: {}
status:
  lastUpdated: "2020-10-28T05:58:42Z"
  namespaces:
  - ""
```
The reason I'm expecting it to be available across all namespaces is the way [`installModes` are configured in the CSV](https://github.com/operator-framework/community-operators/blob/master/community-operators/service-binding-operator/0.1.1-364/service-binding-operator.v0.1.1-364.clusterserviceversion.yaml#L181-L189).

I could be wrong in my understanding.

pedjak replied:
Not sure what do you mean installed across all namespaces? Technically, operator gets installed in a given namespace (e..g `operatrors`) but it watches for resources in all namespaces.

dharmit replied:
> Technically, operator gets installed in a given namespace (e..g operatrors) but it watches for resources in all namespaces.

Thanks for this info. I was, tbh, not aware of this. I've opened https://github.com/openshift/odo/issues/4191 to ensure we don't miss this from odo's side.

At the moment, odo requires an Operator to be installed in current working ns to be able to spin up a service from its CR.

kadel commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,204 @@
+
+In this guide, we will discuss installing two Operators on a minikube environment:
+
+. etcd Operator
```
why etcd? Shouldn't this be a little bit more generic? I don't expect that a lot of odo users will use etcd.

dharmit replied:
Because our example on the [odo.dev](https://odo.dev/docs/operator-hub/#linking-an-odo-component-with-an-operator-backed-service) website explains the process using an etcd Operator.

kadel commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,204 @@
++
+[source,sh]
+----
+$ kubectl create -f crds/apps.openshift.io_servicebindingrequests_crd.yaml
```
could we just create it straight from github without cloning the whole source code?
```
kubectl create -f https://raw.githubusercontent.com/redhat-developer/service-binding-operator/v0.1.1-364/deploy/crds/apps.openshift.io_servicebindingrequests_crd.yaml
```

dharmit replied:
Sure. :+1:

kadel commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,204 @@
++
+[source,sh]
+----
+$ kubectl create -f operator.yaml
```
could we create one liner from this? 
```
curl -sL https://raw.githubusercontent.com/redhat-developer/service-binding-operator/v0.1.1-364/deploy/operator.yaml | \ 
  sed "s/REPLACE_IMAGE/quay.io\/redhat-developer\/app-binding-operator:0799ba7/" | \
  kubectl create -f -
```

dharmit replied:
Why not? I had assumed otherwise. However, what about Windows users? I'm not sure if this would work on Powershell.

kadel reviewed (commented):
I've also noticed that in the section where we describe `odo link` (in operator-hub.adoc) we don't mention that the SBO is required in order for the link to work

kadel commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,204 @@
+EOF
+----
+
+. Next, we modify the `operator.yaml` file to install the Operator in a specific namespace and use the container image that matches the version of the git commit (`git log -n1 --oneline`) of the repository. Make change as per below `git diff` but make sure to set the value of `namespace` to the namespace where you want to install the Operator:
```
Instead of messing with commit hashes users can simply use the release tag.
for example, quay.io/redhat-developer/app-binding-operator:v0.1.1-364

viccuad commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,204 @@
+Discussing this topic is out of scope of this guide so we have stated it as a note.
+====
+
+Enabling the `olm` addon will, among other things, create an `OperatorGroup` in the `operators` namepsace. Make sure that it's there:
```
```suggestion
Enabling the `olm` addon will, among other things, create an `OperatorGroup` in the `operators` namespace. Make sure that it's there:
```

sbose78 commented:
I've added a singe line non-OLM-based installation in https://github.com/redhat-developer/service-binding-operator/releases/tag/v0.1.1-364 .

Everyone, let me know your thoughts.

sbose78 commented:
Folks, Do you need Service Binding to be installed as an OLM operator or would a non-OLM-installation suffice for `odo` as above ^ ?
If `odo` _needs_ Service Binding installed as OLM operator, then the finest experience would be possible only after Service Binding is added to operatorhub.io .

pedjak commented:
> I've added a singe line non-OLM-based installation in https://github.com/redhat-developer/service-binding-operator/releases/tag/v0.1.1-364 .

Thanks @sbose78 - we should really verify with the acceptance tests if this works. before announcing on GitHub. Did you try to run some tests? I see in yaml that we do not specify any namespaces, and we do not specify any namespace as well in `kubectl apply` command. With that, using the provided yaml file, service binding operator would watch only the `default` namespace. We should definitely document that, or make it configurable.

sbose78 commented:
Hi!

It doesn't go into `default`necessarily, it goes into the current namespace context the user's kubeconfig is at ( standard operating procedure, generally) but we could make it clearer. 

No, I didn't run acceptance tests, I pretty much took the validated documentation that's present here and put that in a single yaml. I've validated it manually though

* Ensured the controller comes up error free.
* Ensured that binding works. 


Most of the issues we've seen around installation has been with the rbac setup. If the rbac setup works and we use the correct image sha ( we should cross-verify ), we should be good.

If @dharmit certifies it, I would propose we go ahead with it. If the e2e tests bring up more issues, let's fix the yaml.

pedjak commented:
> It doesn't go into defaultnecessarily, it goes into the current namespace context the user's kubeconfig is at ( standard operating procedure, generally) but we could make it clearer.

  Yes, we should. For example - should SBO installed this way watches for all namespace or only the one that was the current one at the installation time?

kadel commented:
> Folks, Do you need Service Binding to be installed as an OLM operator or would a non-OLM-installation suffice for `odo` as above ^ ?
> If `odo` _needs_ Service Binding installed as OLM operator, then the finest experience would be possible only after Service Binding is added to operatorhub.io .

@sbose78 
The operator would be better but is not a requirement. If users can just copy/paste one or multiple **SIMPLE** `kubectl apply` commands it is enough for now.
Currently is quite a hassle, as you can see in this document :-/

sbose78 commented:
@pedjak Great point, in a non-OLM installation, we probably need to let users know that you can either watch all namespaces "`WATCH_SPACES=""` or the namespace which the operator in installed. With OLM, that experience is taken care of differently.

@kadel Gothca, does the odo code look for SBO's csv or does it directly try using `ServiceBindingRequest`?

dharmit commented:
@kadel 

> I've also noticed that in the section where we describe odo link (in operator-hub.adoc) we don't mention that the SBO is required in order for the link to work

This sucks and must be fixed. I'll make necessary changes to mention that odo relies on SBO for linking. 


@sbose78 
> If `odo` _needs_ Service Binding installed as OLM operator, then the finest experience would be possible only after Service Binding is added to operatorhub.io .

non-OLM installation should be just fine. That's how I installed and tested things to be working on a minikube setup.

> Great point, in a non-OLM installation, we probably need to let users know that you can either watch all namespaces "`WATCH_NAMESPACES=""` or the namespace which the operator in installed. With OLM, that experience is taken care of differently.

May I suggest/request it to be available across all the namespaces till the time it's made available on operatorhub.io, and also while we figure out more around #4191? Only if it makes sense, that is.

dharmit commented:
> I've added a singe line non-OLM-based installation in https://github.com/redhat-developer/service-binding-operator/releases/tag/v0.1.1-364 .
> 
> Everyone, let me know your thoughts.

Could you please modify the YAML to be similar to what I've attached in this comment [install-v0.1.1-364.modified.txt](https://github.com/openshift/odo/files/5492518/install-v0.1.1-364.modified.txt)?

With the ClusterRole in original YAML, it complains being unable to list SBRs at cluster level. I tried to install it in the `default` namespace.

pedjak commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,229 @@
+
+At the moment, to make Service Binding Operator work on Kubernetes, we need to keep a couple of things in mind:
+
+. It needs to be installed by cloning its GitHub repository
```
I think we do not need that, because we can refer directly all yaml files, demonstrated in line 128.

dharmit replied:
I think this will be taken care of by https://github.com/openshift/odo/pull/4140#issuecomment-729427417

pedjak commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,229 @@
+[source,sh]
+----
+$ kubectl create -f https://raw.githubusercontent.com/redhat-developer/service-binding-operator/v0.1.1-364/deploy/crds/apps.openshift.io_servicebindingrequests_crd.yaml
+$ kubectl create -f service_account.yaml -f role.yaml -f role_binding.yaml
```
we can refer these files as well directly from GH, without clone the repo locally.

dharmit replied:
I think this will be taken care of by https://github.com/openshift/odo/pull/4140#issuecomment-729427417

dharmit commented:
> > I've added a singe line non-OLM-based installation in https://github.com/redhat-developer/service-binding-operator/releases/tag/v0.1.1-364 .
> > Everyone, let me know your thoughts.
> 
> Could you please modify the YAML to be similar to what I've attached in this comment [install-v0.1.1-364.modified.txt](https://github.com/openshift/odo/files/5492518/install-v0.1.1-364.modified.txt)?
> 
> With the ClusterRole in original YAML, it complains being unable to list SBRs at cluster level. I tried to install it in the `default` namespace.

@sbose78 I'm not sure if you changed the yaml or not but on a fresh minikube VM, this one-liner just worked! :man_shrugging: 

I'm going to update the PR to use this one-liner instead of the myriad of commands I've wrote :smiley:

Avni-Sharma commented on docs/public/operators-on-minikube.adoc:
```diff
@@ -0,0 +1,127 @@
+NAME                                        READY   STATUS    RESTARTS   AGE
+service-binding-operator-65745c4bdc-gc6km   1/1     Running   0          34m
+
+$ kubectl logs service-binding-operator-65745c4bdc-gc6km --follow
```
the pod hash will keep changing so instead we can check for the deployment of sbo `kubectl rollout status -w deployment/service-binding-operator -n operators`

dharmit replied:
1. I don't think `kubectl rollout` would show the logs of the pod, would it? Reason for asking pod logs is that it would/could help us determine if something's wrong with how SBO is deployed.
1. We're not mentioning to install this in `operators` namespace. So far, things have been tricky when I have tried to install SBO in `operators` ns. For example, before replying to this comment, I tried to install SBO in `operators` ns on a fresh minikube instance. Here's the logs:
    ```
    {"level":"info","ts":1605702505.1782641,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"servicebindingrequest-controller","source":"kind source: operators.coreos.com/v1alpha1, Kind=ClusterServiceVersion"}
    {"level":"error","ts":1605702505.1828096,"logger":"mapper","msg":"listing SBRs","Object.Namespace":"operators","Object.Name":"service-binding-operator-lock","error":"servicebindingrequests.apps.openshift.io is forbidden: User \"system:serviceaccount:operators:service-binding-operator\" cannot list resource \"servicebindingrequests\" in API group \"apps.openshift.io\" at the cluster scope","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/redhat-developer/service-binding-operator/pkg/log.(*Log).Error\n\t/go/src/github.com/redhat-developer/service-binding-operator/pkg/log/log.go:35\ngithub.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest.(*sbrRequestMapper).Map\n\t/go/src/github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/mapper.go:143\nsigs.k8s.io/controller-runtime/pkg/handler.(*EnqueueRequestsFromMapFunc).mapAndEnqueue\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/handler/enqueue_mapped.go:67\nsigs.k8s.io/controller-runtime/pkg/handler.(*EnqueueRequestsFromMapFunc).Create\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/handler/enqueue_mapped.go:47\nsigs.k8s.io/controller-runtime/pkg/source/internal.EventHandler.OnAdd\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/sigs.k8s.io/controller-runtime/pkg/source/internal/eventsource.go:74\nk8s.io/client-go/tools/cache.(*processorListener).run.func1.1\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:658\nk8s.io/apimachinery/pkg/util/wait.ExponentialBackoff\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:292\nk8s.io/client-go/tools/cache.(*processorListener).run.func1\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:652\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88\nk8s.io/client-go/tools/cache.(*processorListener).run\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/client-go/tools/cache/shared_informer.go:650\nk8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1\n\t/go/src/github.com/redhat-developer/service-binding-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71"}
    {"level":"info","ts":1605702505.2789488,"logger":"controller-runtime.controller","msg":"Starting Controller","controller":"servicebindingrequest-controller"}
    {"level":"info","ts":1605702505.2789927,"logger":"controller-runtime.controller","msg":"Starting workers","controller":"servicebindingrequest-controller","worker count":1}
    ```
1. I have mentioned in the doc that pod name will be different from what's shown in the example output.

Avni-Sharma replied:
aha , missed `note that `Pod` name will be different in your environment than what's shown below` In that case resolving the convo

Avni-Sharma commented:
The steps look good to me

kadel commented:
/lgtm
/approve
